src/actions/onboard_actions.js

```diff
@@ -0,0 +1,59 @@
+/*
+Copyright: Ambrosus Technologies GmbH
+Email: tech@ambrosus.com
+
+This Source Code Form is subject to the terms of the Mozilla Public License, v. 2.0. If a copy of the MPL was not distributed with this file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+This Source Code Form is “Incompatible With Secondary Licenses”, as defined by the Mozilla Public License, v. 2.0.
+*/
+
+import {ATLAS, HERMES, APOLLO, ROLE_REVERSE_CODES} from '../consts';
+
+export default class OnboardActions {
+  constructor(kycWhitelistWrapper, rolesWrapper) {
+    this.kycWhitelistWrapper = kycWhitelistWrapper;
+    this.rolesWrapper = rolesWrapper;
+  }
+
+  async onboardAsAtlas(address, url, stakeAmount) {
+    await this.validateWhitelistedForRole(address, ATLAS);
+    await this.validateStakeDepositAmount(address, stakeAmount);
+    const roles = this.rolesWrapper;
+    await roles.onboardAsAtlas(address, url, stakeAmount);
```

Comment:
it might be a good idea to return transaction result in case we will want to display txHash or something

Reply:
I would argue that:
1) YAGNI
2) We are now trying to abstract as much of web3 as possible. In this context: just returning a web3 transaction is not a step into the right direction. Instead we would have to specify (and test) what exactly this methods are supposed to return.

I would therefor take a "implement if needed approach".